native-provider-ci/src/workflows.ts

```diff
@@ -511,9 +511,23 @@ export class PrerequisitesJob implements NormalJob {
   steps: NormalJob["steps"];
   name: string;
   if: NormalJob["if"];
+  permissions?: NormalJob["permissions"];
 
   constructor(name: string, opts: WorkflowOpts) {
     this.name = name;
+    const awsCredentialSteps = steps.ConfigureAwsCredentialsForTests(
+      opts.provider == "aws-native"
+    );
+    if (awsCredentialSteps.length > 0) {
+      // If you specify _any_ permissions then by default the rest of the permissions
+      // that you don't specify will be set to "none"
+      this.permissions = {
+        // Pulumi esc auth requires 'id-token' permission
+        "id-token": "write",
+        // commont-on-pr requires 'pull-requests' permission
```

Comment:
```suggestion
        // comment-on-pr requires 'pull-requests' permission
```

Was this just something we overlooked in a previous PR so the commenting on the PR was just broken, or is this a new requirement of the ESC auth stuff?

Reply:
Oh I think, this is sort of answered by the first comment in the block. `"pull-requests": "write"` is the default, but if we set any permission explicitly, we have to specify everything we need and not just rely on the default. Tricky!